service/role_perm_service.go

```diff
@@ -48,6 +52,10 @@ func (s *RolePermService) GetAllPerms() ([]model.Permission, error) {
 	return s.repo.RolePerm.GetAllPerms()
 }
 
-func (s *RolePermService) AddRoleToUser(user *authProto.User) error {
-	return s.repo.AddRoleToUser(user)
+func (s *RolePermService) AddRoleToUser(user *authProto.User) (*authProto.ResultBinding, error) {
+	err := s.repo.AddRoleToUser(user)
+	if err != nil {
+		return &authProto.ResultBinding{Result: false}, err
```

Comment:
why service converts the result to object from proto? Looks like controller should do it, not service

Reply:
fixed